### PR TITLE
Upgrade LightClient DB

### PR DESCRIPTION
**Pull request by Inspector-Butters**
What type of PR is this?
Feature

What does this PR do? Why is it needed?
This PR upgrades the kv db for light clients to support capella and deneb.

Which issues(s) does this PR fix?
 this is needed for #12991

### CODE REVIEW

**Comment by Inspector-Butters** [comment]
@rkapka it seems that the DB already works ok with different versions of headers. I just added some tests and helper functions.

File: proto/eth/v2/custom.go [hunk 1]
```diff
@@ -65,3 +66,32 @@ func (x *LightClientHeaderContainer) GetBeacon() (*v1.BeaconBlockHeader, error)
 		return nil, fmt.Errorf("unknown header type: %T", input)
 	}
 }
+
+func (x *LightClientHeaderContainer) GetExecutionHeaderCapella() (*enginev1.ExecutionPayloadHeaderCapella, error) {
+	switch input := x.Header.(type) {
+	case *LightClientHeaderContainer_HeaderCapella:
+		return input.HeaderCapella.Execution, nil
+	default:
+		return nil, fmt.Errorf("header type not capella: %T", input)
```
Comment (rkapka) [inline proto/eth/v2/custom.go:75]:
```suggestion
		return nil, fmt.Errorf("header type %T not Capella", input)
```
Semicolons separate errors in Go so it's a best practice not to used them within error messages

File: proto/eth/v2/custom.go [hunk 2]
```diff
@@ -65,3 +66,32 @@ func (x *LightClientHeaderContainer) GetBeacon() (*v1.BeaconBlockHeader, error)
...
+	}
+}
+
+func (x *LightClientHeaderContainer) GetExecutionHeaderDeneb() (*enginev1.ExecutionPayloadHeaderDeneb, error) {
+	switch input := x.Header.(type) {
+	case *LightClientHeaderContainer_HeaderDeneb:
+		return input.HeaderDeneb.Execution, nil
+	default:
+		return nil, fmt.Errorf("header type not deneb: %T", input)
```
Comment (rkapka) [inline proto/eth/v2/custom.go:84]:
```suggestion
		return nil, fmt.Errorf("header type %T not Deneb", input)
```

File: proto/eth/v2/custom.go [hunk 3]
```diff
@@ -65,3 +66,32 @@ func (x *LightClientHeaderContainer) GetBeacon() (*v1.BeaconBlockHeader, error)
...
+}
+
+func (x *LightClientHeaderContainer) GetExecutionBranch() ([][]byte, error) {
+	switch input := x.Header.(type) {
+	case *LightClientHeaderContainer_HeaderCapella:
+		return input.HeaderCapella.ExecutionBranch, nil
+	case *LightClientHeaderContainer_HeaderDeneb:
+		return input.HeaderDeneb.ExecutionBranch, nil
+	default:
+		return nil, fmt.Errorf("wrong header type: %T", input)
```
Comment (rkapka) [inline proto/eth/v2/custom.go:95]:
```suggestion
		return nil, fmt.Errorf("wrong header type %T", input)
```

**Review by rkapka — APPROVED** [review]

**Pull request merged**